src/test/test-lwm2m.c

```diff
@@ -460,7 +1169,7 @@ observe_res_cb(void *data,
     enum sol_lwm2m_content_type content_type,
     struct sol_str_slice content)
 {
-    static int i = 0;
+    static int nosec_i = 0, sec_i = 0;
```

Comment:
Which states would those be? I must confess I didn't "deep dive" in this part - the sequence of tests performed originally, which apparently includes a step of modifying one integer resource and so on. I just changed the `static i` to `static "i(s)"` to support two different clients independently.